### PR TITLE
first pass at adding build/packaging details for Python functions

### PR DESCRIPTION
**Pull request by jazwiecki**
*Issue #, if available:* #2478

*Description of changes:* Adds barebones documentation of build & package steps for Python functions, specifically around packaging dependencies.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/cli/function/build-options.md [hunk 1]
```diff
@@ -3,6 +3,16 @@ title: Build options
 description: Use build options for Amplify's function category to execute a script before a function is deployed, e.g. to transpile Typescript or ES6 with Babel into a format that is supported by the AWS Lambda's node runtime.
 ---
 
+## Python
```
Comment (renebrandel) [inline docs/cli/function/build-options.md:6]: I'd move the Python section below the NodeJS section.

File: docs/cli/function/build-options.md [hunk 2]
```diff
@@ -3,6 +3,16 @@ title: Build options
...
+
+**There are no existing build options for Python functions**
```
Comment (renebrandel) [inline docs/cli/function/build-options.md:8]: Remove the bold lettering here and move it and collapse it into the next paragraph as the first sentence.

File: docs/cli/function/build-options.md [hunk 3]
```diff
@@ -3,6 +3,16 @@ title: Build options
...
+
+The process of building and packaging Python functions is in line with Amazon's [existing documentation](https://docs.aws.amazon.com/lambda/latest/dg/python-package.html#python-package-venv) for manually creating a Lambda deployment package which depends on a virtual environment. Amplify will run `pipenv install` in your function's source directory during builds, using either Pipenv's default virtual environment, or whichever virtual environment happens to be active. Then, during the packaging stage, the contents of the `site-packages` directory for that virtual environment will be zipped up along with the function-specific files.
```
Comment (renebrandel) [inline docs/cli/function/build-options.md:10]: Move "Amplify will run..." into its own paragraph.

Comment (renebrandel) [reply docs/cli/function/build-options.md:10]: Remove "comma" from "...directory during builds, using either..."

File: docs/cli/function/build-options.md [hunk 4]
```diff
@@ -3,6 +3,16 @@ title: Build options
...
+
+This means that the contents of the Python build can include local development dependencies (e.g. for testing) in addition to those necessary for your function to run. This also means that packages installed as "editable" (using the `-e` flag) will not be properly packaged, as their only representation in the `site-packages` directory is an `.egg-link` file pointing to the local, editable code of the dependency.
```
Comment (renebrandel) [inline docs/cli/function/build-options.md:12]: Shorten for readability. Remove "This means..." from both sentences. "The contents of..." and "Packages installed as..."

Comment (renebrandel) [reply docs/cli/function/build-options.md:12]: Shorten for clarity: Remove "properly". "... will not be packaged" is also accurate and shortened.

Comment (renebrandel) [inline docs/cli/function/build-options.md:12]: Shorten for readability: "as their only representation in the `site-packages` directory is an `.egg-link` file pointing to the local, editable code of the dependency" --> "as they are represented as an `.egg-link` file pointing to the local, editable code of the dependency."

**Review by renebrandel — CHANGES_REQUESTED** [review]

**Review by renebrandel — APPROVED** [review]

**Comment by mauerbac** [comment]
@jazwiecki , can you fix this merge conflict?

**Comment by jazwiecki** [comment]
@mauerbac yep, think I got it.

**Comment by mauerbac** [comment]
can we get a review @aws-amplify/amplify-cli

**Comment by jazwiecki** [comment]
AFAIK this is just waiting on a review from the CLI team. Not urgent but it was helpful for me to figure this stuff out, might be useful for other people to know how the CLI packages Python lambdas without having to do their own research.

**Review by edwardfoyle — APPROVED** [review]

**Comment by jazwiecki** [comment]
Figured out how to fix the spellcheck failure, sorry about that!

**Comment by ericclemmons** [comment]
Thanks for the updates! Merging this in...

**Pull request merged**